thorlcr/graph/thgraph.cpp

```diff
@@ -2808,6 +2815,9 @@ void CJobBase::endJob()
     }
     if (exceptions && exceptions->ordinality())
         throw exceptions.getClear();
+
+    // reset for next job
+    setProcessAborted(false);
```

Comment:
@mckellyln @jakesmith should this be before the throw?

Reply:
well maybe this is too early to clear it actually.
If an exception is fired there, it could still catch and clear up cleanly ready for the next job, so it should be cleared a bit later.

The last interaction with the job the slave has is at the end of the QueryDone statement.
And the last interaction in the master is at the end of the while (!stopped) loop in CJobManager::run() (before it loops around to listen for the next job).

@mckellyln - I think should move the setProcessAborted(false) to those 2 places instead.

Reply:
I don't think so.  If this throw is not explicitly caught and env setting coreOnAbort is not set to true then we would get a traceback in the log but no core file.  If this call was moved to before the throw, the only difference I think, would be that if its not caught we would always get a core file.

Reply:
ok, I can move it to those two places.  Are those still better than in init() ?

Reply:
logically, yes I think so, because init() is per subgraph, so occurs many times during a job.

Reply:
ok, if it threw and Thor didn't end then I agree it may not get reset properly for the next job.